### PR TITLE
fix(Select): match experimental spec

### PR DESCRIPTION
**Pull request by emyarod**
Closes IBM/carbon-components-react#2026

Closes https://github.com/IBM/carbon-components-react/issues/2037

This PR incorporates feedback from the February 2019 select component audit

Depends on https://github.com/IBM/carbon-components/pull/2114 and https://github.com/IBM/carbon-components/pull/2135

### CODE REVIEW

File: src/components/Select/Select.js [hunk 1]
```diff
@@ -76,23 +74,51 @@ const Select = React.forwardRef(
...
+        </>
+      );
+    })();
+    return (
+      <div className={`${prefix}--form-item`}>
+        <div className={selectClasses}>
+          <label htmlFor={id} className={labelClasses}>
+            {labelText}
+          </label>
+          {componentsX && !inline && helper}
```
Comment (asudoh) [inline src/components/Select/Select.js]: The markup change seems to affect `v9`. Could you please double-check `v9` is not broken?

**Review by asudoh — COMMENTED** [review]
LGTM basically, just one thing - Thanks @emyarod!

File: src/components/Select/Select.js (hunk 1 above)
Comment (emyarod) [reply src/components/Select/Select.js]: it should be fixed now @asudoh

Comment (asudoh) [reply src/components/Select/Select.js]: OK thank you for adding that back! Actually `v9` topic came up as the helper used to be next icon whereas it's next to `<label>` in this PR. Wanted to make sure such move-around doesn't negatively affecting `v9`. Thanks!

Comment (asudoh) [reply src/components/Select/Select.js]: @emyarod You are right, I may have been seeing a stale commit, my apologies!

**Review by tw15egan — APPROVED** [review]
👍

**Comment by emyarod** [comment]
currently in the middle of incorporating changes from IBM/carbon-components#2114 to add legacy support for inline helper text @tw15egan, will likely need another review in a bit

**Comment by tw15egan** [comment]
Okay, just ping me when re-review is needed

**Comment by emyarod** [comment]
@tw15egan https://github.com/IBM/carbon-components/pull/2135 is now ready for review

**Review by aagonzales — COMMENTED** [review]
For disabled, the label and helper text need to be using the color "$disabled-02"

**Comment by tw15egan** [comment]
@aagonzales seems like they are: https://github.com/emyarod/carbon-components/blob/8ef8e3ccd9eedfb1546cdf38b3b0b6b892d55e09/src/components/select/_select.scss#L287

**Comment by aagonzales** [comment]
Oh ok the knob was only changing the field but I guess the labels are controlled by a different knob?

**Review by aagonzales — APPROVED** [review]
LGTM then!

**Comment by emyarod** [comment]
@aagonzales can you double check again? I was still moving newly added changes from vanilla over when you reviewed, but disabling the component should also be changing label and helper text

**Comment by aagonzales** [comment]
@emyarod yup! The knob is changing the label and helper text now so its good!

**Pull request merged**